src/Intacct/Functions/GeneralLedger/JournalEntryLineCreate.php

```diff
@@ -36,7 +36,13 @@ public function writeXml(XMLWriter &$xml)
 
         $xml->writeElement('DOCUMENT', $this->getDocumentNumber());
         $xml->writeElement('ACCOUNTNO', $this->getGlAccountNumber(), true);
-        if ($this->getTransactionAmount()< 0) {
+
+        // If the amount is 0, we need to explicitly specify whether the entry is Debit or Credit
+        if ($this->getTransactionType()) {
+            $xml->writeElement('TR_TYPE', $this->getTransactionType());
+		    $xml->writeElement('TRX_AMOUNT', abs($this->getTransactionAmount()), true);
```

Comment:
are they any other types besides debit and credit?

Reply:
no

Reply:
https://developer.intacct.com/api/general-ledger/journal-entries/#create-journal-entry